src/json/jsonplatform.c

```diff
@@ -0,0 +1,728 @@
+/*
+ * Author: Houman Brinjcargorabi <Houman.Brinjcargorabi@intel.com>
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#define _GNU_SOURCE
+#if !defined(_XOPEN_SOURCE) || _XOPEN_SOURCE < 600
+#define _XOPEN_SOURCE 600 /* Get nftw() and S_IFSOCK declarations */
+#endif
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+
+#if defined(JSONPLAT)
+#include <json-c/json.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#endif
+
+#include "mraa_internal.h"
+
+#if defined(JSONPLAT)
+typedef mraa_result_t (*init_plat_func_t)(json_object*, mraa_board_t*, int);
+
+inline mraa_result_t
+mraa_init_json_platform_get_pin(json_object* jobj, const char* io, const char* key, int index, unsigned int* pin)
+{
+    json_object* jobj_temp = NULL;
+    if (json_object_object_get_ex(jobj, key, &jobj_temp)) {
+        if (!json_object_is_type(jobj_temp, json_type_int)) {
+            syslog(LOG_ERR, "init_json_platform: %s %s at position: %d is not an int", io, key, index);
+            return MRAA_ERROR_INVALID_RESOURCE;
+        }
+        *pin = (unsigned int) json_object_get_int(jobj_temp);
+        return MRAA_SUCCESS;
+    }
+    syslog(LOG_ERR, "init_json_platform: No %s specified for %s at position: %d", key, io, index);
+    return MRAA_ERROR_NO_DATA_AVAILABLE;
+}
+
+inline mraa_result_t
```

Comment:
Needs to be inline static if you want to do that. Because we already have -finline-small-functions let's just not make these inline at all and let the compiler work out the best option.